langchain4j-azure-cosmos-mongo-vcore/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/mongo/vcore/AzureCosmosDbMongoVCoreEmbeddingStore.java

```diff
@@ -0,0 +1,585 @@
+     *                                vector data. We recommend that numLists is set to documentCount/1000 for up to 1 million
+     *                                documents and to sqrt(documentCount) for more than 1 million documents. Using a numLists value
+     *                                of 1 is akin to performing brute-force search, which has limited performance.
+     * @param dimensions              - Number of dimensions for vector similarity. The maximum number of supported dimensions
+     *                                is 2000.
+     * @param m                       - used only for vector -hnsw. The max number of connections per layer (16 by default, minimum value is 2, maximum
+     *                                value is 100). Higher m is suitable for datasets with high dimensionality and/or high
+     *                                accuracy requirements.
+     * @param efConstruction          - used only for vector -hnsw. The size of the dynamic candidate list for constructing the graph (64 by default, minimum
+     *                                value is 4, maximum value is 1000). Higher ef_construction will result in better index
+     *                                quality and higher accuracy, but it will also increase the time required to build the index.
+     *                                ef_construction has to be at least 2 * m.
+     * @param efSearch                - used only for vector -hnsw. The size of the dynamic candidate list for search (40 by default). A higher value provides
+     *                                better recall at the cost of speed.
+     */
+    public AzureCosmosDbMongoVCoreEmbeddingStore(
+            MongoClient mongoClient,
+            String connectionString,
+            String databaseName,
+            String collectionName,
+            String indexName,
+            String applicationName,
+            CreateCollectionOptions createCollectionOptions,
+            Boolean createIndex,
+            String kind,
+            Integer numLists,
+            Integer dimensions,
+            Integer m,
+            Integer efConstruction,
+            Integer efSearch) {
+        if (mongoClient == null && (connectionString == null || connectionString.isEmpty())) {
+            throw new IllegalArgumentException("You need to pass either the mongoClient or " +
+                    "the connectionString required for connecting to Azure CosmosDB Mongo vCore");
+        }
+
+        if (databaseName == null || databaseName.isEmpty() || collectionName == null || collectionName.isEmpty()) {
+            throw new IllegalArgumentException("databaseName and collectionName needs to be provided.");
+        }
+        createIndex = getOrDefault(createIndex, false);
+        this.indexName = getOrDefault(indexName, "defaultIndexAzureCosmos");
+        applicationName = getOrDefault(applicationName, "JAVA_LANG_CHAIN");
```

Comment:
```suggestion
        applicationName = getOrDefault(applicationName, "LangChain4j");
```
Do you mind changing it?

Reply:
Updated the code